spec/unit/persistent_enum_spec.rb

```diff
@@ -0,0 +1,379 @@
+        expect(val.name).to eq(expected_name.to_s)
+      end
+    end
+  end
+
+  context "with extra fields" do
+    let(:members) do
+      {
+        :One   => { count: 1 },
+        :Two   => { count: 2 },
+        :Three => { count: 3 },
+        :Four  => { count: 4 }
+      }
+    end
+
+    shared_examples "acts like an enum with extra fields" do
+      it "has all expected members with expected values" do
+        members.each do |name, fields|
+          ev = model.value_of(name)
+
+          # Ensure it exists and is correctly saved
+          expect(ev).to be_present
+          expect(model.values).to include(ev)
+          expect(model.all_values).to include(ev)
+          expect(ev).to eq(model[ev.ordinal])
+
+          # Ensure it's correctly saved
+          if model.table_exists?
+            expect(model.where(name: name).first).to eq(ev)
+          end
+
+          # and that fields have been correctly set
+          fields.each do |fname, fvalue|
+            expect(ev[fname]).to eq(fvalue)
+          end
+        end
+      end
+    end
+
+    shared_examples "acts like a persisted enum with extra fields" do
+      include_examples "acts like an enum with extra fields"
+    end
+
+    context "providing a hash" do
+      let(:model) do
+        members = members()
+        create_test_model(:with_extra_field, ->(t){ t.string :name; t.integer :count }) do
+          # pre-existing matching, non-matching, and outdated data
+          create(name: "One", count: 3)
+          create(name: "Two", count: 2)
+          create(name: "Zero", count: 0)
+
+          acts_as_enum(members)
+        end
+      end
+
+      it_behaves_like "acts like a persisted enum"
+      it_behaves_like "acts like a persisted enum with extra fields"
+
+      it "keeps outdated data" do
+        z = model.value_of("Zero")
+        expect(z).to be_present
+        expect(model[z.ordinal]).to eq(z)
+        expect(z.count).to eq(0)
+        expect(model.all_values).to include(z)
+        expect(model.values).not_to include(z)
+      end
+    end
+
+    context "using builder interface" do
+      let(:model) do
+        create_test_model(:with_extra_field_using_builder, ->(t){ t.string :name; t.integer :count }) do
+          acts_as_enum([]) do
+            One(count: 1)
+            Two(count: 2)
+            constant!(:Three, count: 3)
+            Four(count: 4)
+          end
+        end
+      end
+
+      it_behaves_like "acts like a persisted enum"
+      it_behaves_like "acts like a persisted enum with extra fields"
+    end
+
+    context "without table" do
+      let(:model) do
+        members = members()
+        create_test_model(:with_extra_field_without_table, nil, create_table: false) do
+          acts_as_enum(members)
+        end
+      end
+
+      it_behaves_like "acts like an enum"
+      it_behaves_like "acts like an enum with extra fields"
+    end
+
+    it "must have attributes that match the table" do
+      expect {
```

Comment:
Rspec's examples used them. If I had to reverse-engineer a reason why, it's that calling methods on the result of a `do...end` looks odd.